src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8250,6 +8250,19 @@ private ErrorPropertySymbol CreateErrorPropertySymbol(ImmutableArray<PropertySym
             }
             else
             {
+                // If we're performing function pointer overload resolution and we're not in an attribute
+                // argument, then we force complete UnmanagedCallersOnly data for all methods to ensure
+                // they are loaded to be considered for calling convention comparison. If we're in an
+                // attribute argument, then it's an error case and overload resolution will treat it as
+                // matching to suppress potentially spurious diagnostics
+                if (!InAttributeArgument && isFunctionPointerResolution)
```

Comment:
>if (!InAttributeArgument && isFunctionPointerResolution) [](start = 16, length = 56)

It feels like this part belongs in Overload resolution. When it needs the information it should request it. The current approach is somewhat fragile, we have to make sure every consumer does this before OverloadResolution is invoked.